optimum/exporters/openvino/__main__.py

```diff
@@ -203,6 +203,8 @@ def main_export(
         do_gptq_patching = quantization_config and quantization_config["quant_method"] == "gptq"
         model_type = config.model_type.replace("_", "-")
 
+        if model_type in {"falcon", "mpt"} and trust_remote_code:
+            trust_remote_code = False
```

Comment:
mpt and falcon export failed if trust_remote_code used due to different in formats of past key values

Reply:
> mpt and falcon export failed if trust_remote_code used due to difference in formats of past key values

I imagine it could be the case for all architectures no ? Also is this only for stateful models ? (should work for MPT at least for the case with and without pkv for non stateful models as I see here for ONNX https://github.com/huggingface/optimum/blob/253c6c2ff0f834b6fb03557add6ba4ae1272714d/tests/exporters/onnx/test_onnx_export.py#L570)

Reply:
I updated condition and added warning for using this models, users still can convert models with trust_remote_code for other tasks or if they provide custom configs